scrub/url.go

```diff
@@ -0,0 +1,85 @@
+package scrub
+
+import (
+	"net/url"
+	"strings"
+)
+
+// The list of URL parameter names that will have their contents scrubbed of sensitive data.
+var (
+	// copy from https://github.com/heroku/rollbar-blanket/blob/master/lib/rollbar/blanket/fields.rb
+	RestrictedParams = map[string]bool{
+		"access_token":                                true,
+		"api_key":                                     true,
+		"authenticity_token":                          true,
+		"body.trace_chain.0.extra.cookies":            true,
+		"body.trace_chain.0.extra.msg":                true,
+		"body.trace_chain.0.extra.session.csrf.token": true,
+		"bouncer.refresh_token":                       true,
+		"bouncer.token":                               true,
+		"confirm_password":                            true,
+		"fingerprint":                                 true,
+		"heroku_oauth_token":                          true,
+		"heroku_session_nonce":                        true,
+		"heroku_user_session":                         true,
+		"key":                                true,
+		"oauth_token":                        true,
+		"old_secret":                         true,
+		"passwd":                             true,
+		"password":                           true,
+		"password_confirmation":              true,
+		"postgres_session_nonce":             true,
+		"private_key":                        true,
+		"request.cookies":                    true,
+		"request.cookies.signup-sso-session": true,
+		"request.params._csrf":               true,
+		"request.session._csrf_token":        true,
+		"request.session.csrf.token":         true,
+		"secret":                             true,
+		"secret_token":                       true,
+		"sudo_oauth_token":                   true,
+		"super_user_session_secret":          true,
+		"token":                              true,
+		"user_session_secret":                true,
+		"www-sso-session":                    true,
+	}
+)
+
+// URL removes a subset of "sensitive" URL parameters as used by parts of Heroku's
+// conventions for API design. The output of this function is safe to be logged
+// except in high-security scenarios.
+func URL(u *url.URL) *url.URL {
```

Comment:
I  wonder if this should take a string instead of a *url.URL. I guess this is most likely to be used with a http.Request, which includes and already parsed *url.URL though, so nevermind.